plugins/module_utils/config_utils.py

```diff
@@ -99,3 +104,192 @@ def changed(self) -> bool:
+        """
+        Retrive list of configure_functions for a given module and setting from the version-specific mapping.
+        """
+
+        map_dict = self.version_map.get(self.version)
+
+        if map_dict is None:
+            raise KeyError(f"{self.version} was not not found in version_map")
+
+        # Check if the provided module is in the map
+        if module in map_dict and setting in map_dict[module]:
+            # Check if the setting contains a dictionary of configure functions
+            if isinstance(map_dict[module][setting], dict):
+                return map_dict[module][setting]
+
+    def set_module_setting(self, value: str, module: str = None, setting: str = None):
+        """
+        utility to set config specific setting
+        """
+
+        # get xpath from key_mapping
+        xpath = self._get_xpath(module=module, setting=setting)
+
+        # create a copy of the _config_dict
+        _config_dict = self._config_dict.copy()
+
+        # iterate over xpath value to get specific key
+        keys = xpath.split("/")
+
+        for key in keys[:-1]:
+            _config_dict = _config_dict.setdefault(key, {})
+
+        # Update the final value
+        _config_dict[keys[-1]] = value
+
+    def del_module_setting(self, module: str = None, setting: str = None):
+        """
+        utility to delete config specific setting
+        """
+
+        # get xpath from key_mapping
+        xpath = self._get_xpath(module=module, setting=setting)
+
+        # create a copy of the _config_dict
+        _config_dict = self._config_dict.copy()
+
+        # iterate over xpath value to get specific key
+        keys = xpath.split("/")
+
+        for key in keys[:-1]:
+            _config_dict = _config_dict.setdefault(key, {})
+
+        # Update the final value
+        _config_dict[keys[-1]] = None
+
+    def get_module_setting(self, module: str = None, setting: str = None) -> str:
+        """
+        utility to get config specific setting
+        """
+
+        # get xpath from key_mapping
+        xpath = self._get_xpath(module=module, setting=setting)
+
+        # create a copy of the _config_dict
+        _config_dict = self._config_dict.copy()
+
+        # iterate over xpath value to get specific key
+        for key in xpath.split("/"):
+            _config_dict = _config_dict[key]
+
+        # return key
+        return _config_dict
+
+    def apply_module_setting(self, module: str = None) -> List[str]:
+        """
+        utility to get and apply config specific php_requirements
+        """
+
+        # get version and module specific php_requirements
+        php_requirements = self._get_php_requirements(module=module, setting="php_requirements")
+
+        # get version and module specific configure_functions
+        configure_functions = self._get_configure_functions(
+            module=module,
+            setting="configure_functions"
+        )
```

Comment:
`_get_configure_functions` might return `None` so it would probably make sense to add some error handling here

Reply:
Thanks for pointing out, I added a check to ensure `configure_function` is not `None`.